google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineCloudType.java

```diff
@@ -316,16 +316,21 @@ public void connect(@NotNull ConnectionCallback<AppEngineDeploymentConfiguration
       ServerRuntimeInstance<AppEngineDeploymentConfiguration> {
 
     private AppEngineServerConfiguration configuration;
+    private AppEngineAction currentAction;
 
     public AppEngineRuntimeInstance(
         AppEngineServerConfiguration configuration) {
       this.configuration = configuration;
     }
 
     @Override
-    public void deploy(@NotNull final DeploymentTask<AppEngineDeploymentConfiguration> task,
```

Comment:
OK, so if it's not scoped to a specific deployment action, is synchronization still the right thing to do? Won't that just block the next deployment?